crates/containerd-shim-wasm/src/sandbox/shim.rs

```diff
@@ -1,1274 +0,0 @@
-            id: "test".to_string(),
-            ..Default::default()
-        })?;
-        assert!(res.has_stats());
-
-        local.task_kill(api::KillRequest {
-            id: "test".to_string(),
-            signal: 9,
-            ..Default::default()
-        })?;
-
-        rx.recv_timeout(Duration::from_secs(5)).unwrap()?;
-
-        let state = local.task_state(api::StateRequest {
-            id: "test".to_string(),
-            ..Default::default()
-        })?;
-        assert_eq!(state.status(), Status::STOPPED);
-
-        local.task_delete(api::DeleteRequest {
-            id: "test".to_string(),
-            ..Default::default()
-        })?;
-
-        match local
-            .task_state(api::StateRequest {
-                id: "test".to_string(),
-                ..Default::default()
-            })
-            .unwrap_err()
-        {
-            Error::NotFound(_) => {}
-            e => return Err(e),
-        }
-
-        Ok(())
-    }
-}
-
-impl<T: Instance + Send + Sync> Local<T> {
-    /// Creates a new local task service.
-    pub fn new(
-        engine: T::Engine,
-        tx: Sender<(String, Box<dyn MessageDyn>)>,
-        exit: Arc<ExitSignal>,
-        namespace: String,
-        containerd_address: String,
-    ) -> Self {
-        Self {
-            // Note: engine.clone() is a shallow clone, is really cheap to do, and is safe to pass around.
-            engine,
-            instances: Arc::new(RwLock::new(HashMap::new())),
-            events: Arc::new(Mutex::new(tx)),
-            exit,
-            namespace,
-            containerd_address,
-        }
-    }
-
-    fn new_base(&self, id: String) -> InstanceData<T> {
-        let cfg = InstanceConfig::new(
-            self.engine.clone(),
-            self.namespace.clone(),
-            self.containerd_address.clone(),
-        );
-        InstanceData {
-            instance: InstanceOption::Nop(Nop::new(id, None).unwrap()),
-            cfg,
-            pid: RwLock::new(None),
-            state: Arc::new(RwLock::new(TaskState::Created)),
-        }
-    }
-
-    fn send_event(&self, event: impl Event) {
-        let topic = event.topic();
-        self.events
-            .lock()
-            .unwrap()
-            .send((topic.clone(), Box::new(event)))
-            .unwrap_or_else(|e| warn!("failed to send event for topic {}: {}", topic, e));
-    }
-
-    fn get_instance(&self, id: &str) -> Result<Arc<InstanceData<T>>> {
-        self.instances
-            .read()
-            .unwrap()
-            .get(id)
-            .ok_or_else(|| Error::NotFound(id.to_string()))
-            .map(|i| i.clone())
-    }
-
-    fn instance_exists(&self, id: &str) -> bool {
-        self.instances.read().unwrap().contains_key(id)
-    }
-
-    fn is_empty(&self) -> bool {
-        self.instances.read().unwrap().is_empty()
-    }
-
-    fn task_create(&self, req: api::CreateTaskRequest) -> Result<api::CreateTaskResponse> {
```

Comment:
It looks like there are some significant changes to this function in the new `local.rs` code. Need a closer look into it...

Reply:
The main difference is that we are not mounting the non-rootfs mounts, i.e., [this for loop](https://github.com/containerd/runwasi/pull/345/files#diff-8f8704e2bb5644be97cfbd99c736cb7f432da2de20292de2b0f69e4193bd0f64L750-L752) is gone.
IIUC, libcontainer does this for us.
But I might be wrong, and would appreciate the extra pair of eyes :-)

Reply:
`libcontainer` mounts everything in `spec.mounts()` in [here](https://github.com/containers/youki/blob/710cb4c636bc693a429aea32f840260f0d2b03b2/crates/libcontainer/src/rootfs/rootfs.rs#L90).